Wikipedia/UI-V5/WMFAppViewController.m

```diff
@@ -362,4 +360,14 @@ - (void)searchLanguageDidChangeWithNotification:(NSNotification*)note {
     [self configureSearchViewController];
 }
 
+#pragma mark - UINavigationControllerDelegate
+
+- (void)navigationController:(UINavigationController*)navigationController
```

Comment:
This logic is a little hidden and not very obvious for the rest of the app (Its not clear who is hiding and unhiding the tool bar). Would be better that this is in a UINavigaitonController subclass where you can override push/pop.